src/sprightly.ts

```diff
@@ -1,3 +1,60 @@
-export = function sprightly() {
-  console.log("sprightly")
+import { readFile } from "fs/promises"
+
+interface Data {
+  [key: string]: string | number | Array<string | number | Data> | Data
 }
+
+type Options = {
+  cache?: boolean
+  keyFallback?: string
+  throwOnKeyNotfound?: boolean
+}
+
+const defaultOptions: Options = {
+  keyFallback: "",
+  throwOnKeyNotfound: false,
+  cache: false,
+}
+
+export = async function sprightly(
+  entryPoint: string,
+  data: Data,
+  options: Options = defaultOptions,
```

Comment:
This handles not passing the options parameter but doesn't handle passing an options object but not all options. You should merge the passed options parameter with the default options object.